### PR TITLE
feat: articles page layout impl EXLM-1209

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1209

Test URLs:

Before: https://main--exlm--adobe-experience-league.hlx.page/en/articles/experience-manager
After: https://feat-exlm-1209--exlm--adobe-experience-league.hlx.live/en/articles/experience-manager

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -280,18 +280,26 @@ export const decorateLinks = (block) => {
...
-  return theme
-    .split(',')
-    .map((t) => t.toLowerCase().trim())
-    .includes(type);
+  return isPageOfType(type);
+}
+
+export function isArticlePage(type = 'articles') {
+  return isPageOfType(type);
 }
```
Comment (noormd08) [inline scripts/scripts.js:304]: We may need to keep single function to handle this scenario, as both of them serving same code base.

Comment (nitin-rachabathuni) [reply scripts/scripts.js:304]: isDocsPage is generic logic Noor, I have updated it into different functions for better readability...

**Review by noormd08 — DISMISSED** [review]

File: scripts/scripts.js (hunk 1 above)
Comment (noormd08) [reply scripts/scripts.js:304]: Fine for now.

Comment (msagolj) [reply scripts/scripts.js:304]: @nitin-rachabathuni your checking for a theme named articles but looking at the metadata spreadsheet, there is no such theme set:
image

File: scripts/scripts.js [hunk 2]
```diff
@@ -959,8 +998,10 @@ async function loadPage() {
   // THIS IS TEMPORARY FOR SUMMIT.
   if (handleHomePageHashes()) return;
   // END OF TEMPORARY FOR SUMMIT.
+  addMiniTocForArticlesPage();
```
Comment (msagolj) [inline scripts/scripts.js:1001]: @nitin-rachabathuni this looks like a typical example of auto-blocking so i would move this to into [buildAutoBlocks](https://github.com/adobe-experience-league/exlm/pull/769/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR186)

Comment (nitin-rachabathuni) [reply scripts/scripts.js]: done @msagolj

File: scripts/scripts.js [hunk 3]
```diff
@@ -804,6 +812,37 @@ async function loadRails() {
...
+    if (articleBody && !articleBody.querySelector('.mini-toc')) {
+      // Dynamically add mini-toc section for articles page
+      const miniTocWrapper = htmlToElement(`
+      <div class="mini-toc">
+        <div>
+          <div></div>
+        </div>
+      </div>
+      `);
+      articleBody.appendChild(miniTocWrapper);
```
[the remark this answers is not in the document]
Comment (nitin-rachabathuni) [reply scripts/scripts.js:842]: done @msagolj

**Pull request merged**